collector/cpu_linux.go

```diff
@@ -83,8 +114,64 @@ func NewStatCollector() (Collector, error) {
 	}, nil
 }
 
+func (c *cpuCollector) Update(ch chan<- prometheus.Metric) error {
+	err1 := c.UpdateStat(ch)
+	err2 := c.UpdateCpufreq(ch)
+	if err1 != nil {
+		return err1
+	}
+	if err2 != nil {
+		return err2
+	}
```

Comment:
I don't think using a for loop helps readability. So far we've mostly followed an all or nothing approach I think and aborted after the first error.

I'm fine with the approach above. You could use more descriptive names than a generic counter like `errStat`, but given the few lines involved it doesn't matter too much.